src/http-adapter/jquery/index.js

```diff
@@ -1,148 +1,84 @@
+   * @param {Object} options - request options to be propagated to the request
+   * @param {string} options.url - the url of the request
+   * @param {HTTP_METHOD} options.method - the method of the request
+   * @param {Object} options.headers - A key/value object of headers to be sent with the request
+   * @param {Object} options.data - A key/value payload to be sent with the request. If method is GET, these
+   * options will be transformed into query paramters
+   * @param {boolean} options.isInternal - Whether a request is being sent to an internal server
+   * or not to invoke request/response interceptors
+   * @returns {Promise} Promise relating to request resolution.
    */
-  destroy (link) {
-    return this._ajax('DELETE', link)
-  }
-
-  _ajax (type, url, data) {
-    let headers = {
-      'Accept': 'application/vnd.api+json',
-      'Content-Type': 'application/vnd.api+json'
-    }
-
+  async _makeRequest ({
+    url,
+    method,
+    headers,
+    data,
+    isInternal
+  }) {
     // Stringify data before any async stuff, just in case it's accidentally a mutable object (e.g.
     // some instrumented Vue data)
-    if (data && ['PATCH', 'POST'].indexOf(type) > -1) {
+    if (data && [HTTP_METHOD.PATCH, HTTP_METHOD.POST].indexOf(method) > -1) {
       data = JSON.stringify(data)
     }
 
-    let promise
-    if (this.serializeRequests) {
-      // Wait for all requests to settle (either with success or rejection) before making request
-      const promises = Array.from(this._outstandingRequests)
-        .map(promise => promise.catch(() => {}))
-
-      promise = Promise.all(promises)
-        .then(() => this._makeRequest({ url, type, headers, data }))
-    } else {
-      promise = this._makeRequest({ url, type, headers, data })
-    }
+    const requestInterceptor = this.requestInterceptor
+    const responseInterceptor = this.responseInterceptor
 
-    this._outstandingRequests.add(promise)
-    const removeFromOutstandingRequests = () => {
-      this._outstandingRequests.delete(promise)
-    }
-    promise.then(removeFromOutstandingRequests, removeFromOutstandingRequests)
-
-    return promise
-  }
-
-  _makeRequest ({ url, type, headers, data }) {
     return new Promise((resolve, reject) => {
       let request = {
         url,
-        type,
+        type: method,
         headers,
-        success: (data, textStatus, jqXhr) => {
+        beforeSend (xhr, options) {
+          if (isInternal) {
+            requestInterceptor(xhr, options)
+          }
+        },
+        success: async (data, textStatus, jqXhr) => {
+          if (isInternal) {
+            await responseInterceptor(jqXhr, textStatus, data)
+          }
+
           if (!data && jqXhr.status !== 204) {
             throw new Error(`request returned ${jqXhr.status} status without data`)
           }
-          return resolve(data)
+          return method !== HTTP_METHOD.DELETE ? resolve(JSON.parse(data)) : resolve(data)
```

Comment:
Is this JSON parsing needed now because we turned off the intelligent response type guessing?

Reply:
Correct. It's the reason we have needed it. The `dataType=text` is pre existing in the old adapter, and I am afraid to take it out 😅